adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -462,17 +491,57 @@ def connect_AP(self, ssid, password): # pylint: disable=invalid-name
             self.wifi_set_passphrase(ssid, password)
         else:
             self.wifi_set_network(ssid)
-        for _ in range(10): # retries
+        times = time.monotonic()
+        while (time.monotonic() - times) < timeout_s:  # wait up until timeout
             stat = self.status
             if stat == WL_CONNECTED:
                 return stat
-            time.sleep(1)
+            time.sleep(0.05)
         if stat in (WL_CONNECT_FAILED, WL_CONNECTION_LOST, WL_DISCONNECTED):
             raise RuntimeError("Failed to connect to ssid", ssid)
         if stat == WL_NO_SSID_AVAIL:
             raise RuntimeError("No such ssid", ssid)
         raise RuntimeError("Unknown error 0x%02X" % stat)
 
+    def create_AP(self, ssid, password, channel=1, timeout_s=10): # pylint: disable=invalid-name
+        """
+        Create an access point with the given name, password, and channel.
+        Will wait until specified timeout seconds and return on success
+        or raise an exception on failure.
+
+        :param ssid: the SSID of the created Access Point. Must be 8 or more characters
+        :param passphrase: the password of the created Access Point. Must be 8 or more characters.
+        :param channel: channel of created Access Point (1 - 14).
+        :param timeout_s: number of seconds until we time out and fail to create AP
+        """
+        if len(ssid) < 8:
+            raise RuntimeError("ssid must be 8 more more characters")
+        if len(password) < 8:
```

Comment:
> A pass-phrase is a sequence of between 8 and 63 ASCII-encoded characters. The limit of 63 comes from the desire to distinguish between a pass-phrase and a PSK displayed as 64 hexadecimal characters.

check that the password is password be ascii-encoded and doesnt go beyond 64 chars.